### PR TITLE
Support user-defined upload policies

### PR DESCRIPTION
**Pull request by barata**
This PR adds support for user-defined upload policies. The upload policy must inherits abstract class UploadPolicy and implement 'shouldUpload(TopicPartition)' method. The subclass to be used as current policy is loaded from 'secor.uploadpolicy.strategy.class' property. Furthermore, an additional property 'secor.uploadpolicy.check.seconds' is used to determine how often the policy should be called.

Should the policy need to read any specific property, it can use 'secor.uploadpolicy.properties'. Properties are in the form key1=value1,key2=value2.

For example:

```
secor.uploadpolicy.check.seconds=600
secor.uploadpolicy.strategy.class=com.pinterest.secor.uploader.DefaultUploadPolicy
secor.uploadpolicy.properties=MAX_FILE_SIZE_BYTES=200000000,MAX_FILE_AGE_SECONDS=3600
```

In short, default values do not modify current Secor behavior. Therefore, DefaultUploadPolicy#shouldUpload() returns true if (size >= MAX_FILE_SIZE_BYTES OR modificationAgeSec >= MAX_FILE_AGE_SECONDS).

The following two properties were removed in order to use the new introduced model:

```
secor.max.file.size.bytes
secor.max.file.age.seconds
```

### CODE REVIEW

**Comment by barata** [comment]
@pgarbacki Could you please review my PR? It was motivated by some requirements from my company. We need a special policy that ensures events from Kafka do not take longer than X minutes to be uploaded to S3. That is why I also created a TimeBasedUploadPolicy.
Thanks a lot.

**Comment by fabeschan** [comment]
secor.uploadpolicy.check.seconds is a useful addition. the current check period could result in creating vastly larger file sizes than specified in secor.max.file.size.bytes whenever the amount of data flow is large

**Comment by barata** [comment]
@HenryCaiHaiying Do you think this is relevant? Can we proceed on this?

**Comment by barata** [comment]
Thanks @alshamiri2 . Do you want me to rebase this PR? Do you have any plan to merge it?

**Comment by HenryCaiHaiying** [comment]
This PR would need to sync with the latest code.

**Comment by barata** [comment]
Sync'ing is not a problem. I can do if you want. Just let me know if you find this PR useful.

**Comment by HenryCaiHaiying** [comment]
There are quite a bit of change in the upload policy, take a look to see whether that satisfies your needs.  I don't quite like the idea of using this parameter format:
secor.uploadpolicy.properties=MAX_FILE_SIZE_BYTES=200000000,MAX_FILE_AGE_SECONDS=3600

The old way seems fine with me:
secor.max.file.size.bytes
secor.max.file.age.seconds
